src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/CommunityAPIBookingService.kt

```diff
@@ -101,10 +101,13 @@ class CommunityAPIBookingService(
       .toString()
   }
 
-  fun isRescheduleBooking(existingAppointment: Appointment, appointmentTime: OffsetDateTime, durationInMinutes: Int): Boolean =
-    isDifferentTimings(existingAppointment, appointmentTime, durationInMinutes)
+  fun isRescheduleBooking(existingAppointment: Appointment, appointmentTime: OffsetDateTime, durationInMinutes: Int, npsOfficeCode: String?): Boolean =
+    isDifferentTimings(existingAppointment, appointmentTime, durationInMinutes) || isDifferentLocation(existingAppointment, npsOfficeCode)
 
-  fun isDifferentTimings(existingAppointment: Appointment, appointmentTime: OffsetDateTime, durationInMinutes: Int): Boolean =
+  private fun isDifferentLocation(existingAppointment: Appointment, npsOfficeCode: String?): Boolean {
+    return !npsOfficeCode.equals(existingAppointment.appointmentDelivery?.npsOfficeCode)
```

Comment:
npsOfficeCode is defined as nullable - should this check be made safe

Reply:
the only place npsOfficeCode should not be null is on the main method beacuse then it gets defaulted?

Reply:
Sorry, I a bit unsure of the question.

`npsOfficeCode` being null means the appointment was either a phone call, video call or custom office location.
`npsOfficeCode` not being null means the appointment was in a defined PP office.


I've changed the assignment of the default `npsOfficeCode` to only be set when defining the DTO request to send to delius.
We do not store the default npsOfficeCode in the database when it's a non-PP office. So I need to have this method return true when the appointment is a phone call and the appointment change is also a phone call. In which case they would both be null.

Perhaps the variable would make more sense to be called: `userSuppliedNpsOfficeCode` rather than `npsOfficeCodeToSubmitToDelius`.

Reply:
LGTM